apps/taxes/src/modules/avatax/avatax-customer-code-resolver.ts

```diff
@@ -1,18 +1,32 @@
-import { OrderConfirmedSubscriptionFragment } from "../../../generated/graphql";
-import { CalculateTaxesPayload } from "../../pages/api/webhooks/checkout-calculate-taxes";
+import { createLogger } from "@saleor/apps-logger";
+import { UserFragment } from "../../../generated/graphql";
 
-export class AvataxCustomerCodeResolver {
-  /*
-   * "If you're creating a sales order for a customer that does not yet exist in your system, you can send a dummyCustomerCode as sales orders are not recorded."
-   * https://developer.avalara.com/ecommerce-integration-guide/sales-tax-badge/transactions/simple-transactions/
-   */
-  private fallbackCustomerCode = "0";
+const logger = createLogger("avataxCustomerCode");
 
-  resolveOrderCustomerCode(order: OrderConfirmedSubscriptionFragment) {
-    return order.user?.id ?? this.fallbackCustomerCode;
-  }
+/*
+ * "If you're creating a sales order for a customer that does not yet exist in your system, you can send a dummyCustomerCode as sales orders are not recorded."
+ * https://developer.avalara.com/ecommerce-integration-guide/sales-tax-badge/transactions/simple-transactions/
+ */
+const FALLBACK_CUSTOMER_CODE = "0";
 
-  resolveCalculateTaxesCustomerCode(payload: CalculateTaxesPayload): string {
-    return payload.taxBase.sourceObject.user?.id ?? this.fallbackCustomerCode;
-  }
-}
+export const avataxCustomerCode = {
+  resolve(user: UserFragment | null | undefined): string {
+    // see: docs.saleor.io/docs/3.x/developer/app-store/apps/taxes/avatax/overview#customer-code
+    const avataxCustomerCode = user?.avataxCustomerCode;
+
+    if (avataxCustomerCode) {
+      logger.trace(`Returning customer code found in the user metadata.`);
```

Comment:
suggestion: Let's add the customer code to logs for debugging purposes

```suggestion
      logger.trace({ avataxCustomerCode }, `Returning customer code found in the user metadata.`);
```